mono/metadata/threads.c

```diff
@@ -1614,9 +1619,8 @@ MonoString*
 ves_icall_System_Threading_Thread_GetName_internal (MonoInternalThread *this_obj)
 {
 	MonoError error;
-	MonoString* str;
-
 	error_init (&error);
+	MonoString* str;
```

Comment:
This seems unmotivated.

(and this icall is due for a conversion to use handles - In fact we had a previous PR #5725 that attempted it that is on my todo list to revive and finish).

Reply:
Point was to always error_init immediately after MonoError error; I can undo for now but it should be done some way or another.